packages/bpk-component-price/src/BpkPrice.js

```diff
@@ -0,0 +1,117 @@
+/*
+ * Backpack - Skyscanner's Design System
+ *
+ * Copyright 2016 Skyscanner Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/* @flow strict */
+
+import PropTypes from 'prop-types';
+import React from 'react';
+import { cssModules } from 'bpk-react-utils';
+import BpkBreakpoint, { BREAKPOINTS } from 'bpk-component-breakpoint';
+import BpkText, { TEXT_STYLES } from 'bpk-component-text';
+
+import STYLES from './BpkPrice.module.scss';
+
+const getClassName = cssModules(STYLES);
+
+export type Props = {
+  title: string,
+  className: ?string,
+  subtitle: ?string,
+  description: ?string,
+};
+
+const BpkPrice = (props: Props) => {
+  const { className, description, subtitle, title, ...rest } = props;
+
+  return (
+    <BpkBreakpoint query={BREAKPOINTS.MOBILE}>
+      {(isMobile) =>
+        isMobile ? (
+          <div
```

Comment:
A lot of this code is duplicated across when it's mobile or desktop. Could we check to see if we can simplify this instead

For example using isMobile when setting specific class names or !isMobile for when wanting to set classes inside the getClassNames function

Reply:
As @colorwin said [here](https://skyscanner.slack.com/archives/C03GU582NQP/p1654164733818369), it would be more flexible to use a param to control this layout, which would at least have the following benefits
1. More flexible usages
2. No need to add `BpkBreakpoint` as a dependency to reduce the package size
3. It will not support SSR very well, because `BpkBreakpoint` does not support it, that is, if we render this component on the server side, it will be re-rendered on the client side causing to flicker, but if we use param instead, we can use our [custom component](https://github.skyscannertools.net/hotels-web/hotels-website/blob/main/packages/client/src/components/Common/BpkBreakpoint/BpkBreakpointWrapper.tsx#L22) to solve this issue